apps/logging/loggers.py

```diff
@@ -1,10 +1,29 @@
 import json
 import logging
 
-from django.conf import settings
-
 from apps.dot_ext.models import get_application_counts, get_application_require_demographic_scopes_count
 from apps.fhir.bluebutton.models import check_crosswalks
+from django.conf import settings
+
+
+"""
+  Logger and logging function for waffle flags, switches
+"""
+waffle_event_logger = logging.getLogger('audit.waffle.event')
+
+
+def log_v2_blocked(user=None, path=None, app=None, err=None, **kwargs):
+    log_dict = {"type": "v2_blocked",
+                "user": str(user) if user else None,
+                "path": path if user else None,
+                "app_id": app.id if app else None,
+                "app_name": str(app.name) if app else None,
+                "dev_id": str(app.user.id) if app else None,
+                "dev_name": str(app.user.username) if app else None,
+                "response_code": err.status_code,
+                "message": str(err) if err else None}
```

Comment:
Is this the standard way for logging out an error message? via the `message` property? It's nice to stick to a defined pattern so that when we are searching for all errors - this one will come up.

Reply:
I am not aware of any hard rules about the structure of a BB2 logging record yet, but it seems that the log record be:
1) a valid json string
2) there is a type and all other attributes are up to the intension of how the record is to be consumed e.g. by splunk.

here the "message" is not actually needed by the consumer - "V2 Activities Dashboard", here the type is enough to determine the nature of the event, but when viewed by human reader e..g during a splunk search, that message might explain more ....

may be down the road, there could be some high level schema applied to all BB2 log records to give them a uniformed look...

Reply:
That makes sense. We can leave it as is for now, and maybe with some of the future logging tickets we'll create a standard format.